src/measuro.hpp

```diff
@@ -2109,6 +2109,242 @@ namespace measuro
 
     };
 
+    /*!
+     * @class PrometheusRenderer
+     *
+     * @brief Renders metrics in the Prometheus text-based exposition format.
+     *
+     * Rendered metrics consist of
+     *
+     * Reference: https://github.com/prometheus/docs/blob/main/content/docs/instrumenting/exposition_formats.md#text-based-format
+     *
+     * @include prometheus_renderer_example.txt
+     *
+     * @remarks thread-hostile
+     */
+    class PrometheusRenderer : public Renderer
+    {
+    public:
+        /*!
+         * Constructor.
+         *
+         * @param[in]    destination    Output stream to which to write the rendered metrics. This stream object @b must persist beyond the life of the renderer
+         *
+         * @remarks thread-hostile
+         */
+        PrometheusRenderer(std::ostream & destination, std::function<std::int64_t ()> timestamp_getter,
+                const std::string & app_name = "") noexcept : m_destination(destination), m_timestamp_getter(timestamp_getter),
+                m_count(0), m_app_name(app_name), m_METRIC_NAME_PATTERN("[a-zA-Z_:][a-zA-Z0-9_:]*"),
+                m_METRIC_NAME_CHAR_PATTERN("[a-zA-Z0-9_:]"), m_METRIC_UNIT_CHAR_PATTERN("[a-zA-Z0-9]")
```

Comment:
Wouldn't these constants be better as `static inline const` ?

Reply:
This is `[only supported from C++17](https://en.cppreference.com/w/cpp/language/inline)`, but Measuro targets >= C++14